interfaces/policy/policy.go

```diff
@@ -116,22 +116,23 @@ func (ic *InstallCandidate) Check() error {
 
 // ConnectCandidate represents a candidate connection.
 type ConnectCandidate struct {
-	// TODO: later we need to carry dynamic attributes once we have those
 	Plug                *snap.PlugInfo
 	PlugSnapDeclaration *asserts.SnapDeclaration
+	PlugAttrs           map[string]interface{}
 
 	Slot                *snap.SlotInfo
 	SlotSnapDeclaration *asserts.SnapDeclaration
+	SlotAttrs           map[string]interface{}
 
 	BaseDeclaration *asserts.BaseDeclaration
 }
 
 func (connc *ConnectCandidate) plugAttrs() map[string]interface{} {
-	return connc.Plug.Attrs
+	return connc.PlugAttrs
```

Comment:
there's just a couple of places where is used for real in ifacestate so the risk of getting it wrong there assuming good tests there should not be big, either way we would need test to check that those fields are set because if they are left empty we get would get wrong behavior either way